Parser/Node.cs

```diff
@@ -44,12 +55,43 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
 
+// A Read statement
+public record NReadStmt (Token Name, Token[] Names) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
 // An assignment statement
 public record NAssignStmt (Token Name, NExpr Expr) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+// A Call statement
+public record NCallStmt (Token Name, NExpr[] Exprs) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// An IF statement
+public record NIfStmt (NExpr Expr, NStmt[] Stmts) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A WHILE statement
+public record NWhileStmt (NExpr Expr, NStmt Stmt) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A REPEAT statement
+public record NRepeatStmt (NStmt[] Stmts, NExpr Expr) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A FOR statement
+public record NForStmt (Token Name, bool IsTo, NExpr[] Exprs, NStmt NStmt) : NStmt {

```

Comment:
Change this to:
`public record NForStmt (Token Var, NExpr Start, bool Ascending, NExpr End, NStmt Body)`